### ISSUE TITLE
Incorrect casing for “ID”

### ISSUE THREAD

**Issue by toastal** [yesodweb/persistent#1549]
All implicit generated IDs such as the ID for `Foo` is generated as `FooId`. While much of the documentation uses “ID”, the code representation is incorrect casing where “ID” is the standard abbreviation form for identifiers (IDs), not “Id”. Changing this would align other correctly-cased variables in the project like `DB` (along with uses of `IO`, `UUID`, & `FromJSON` from other libs), but still leaves incorrect cases like `Sql` instead of `SQL` for the initialism ‘structured query language’ (where documentation is largely using “SQL”).

What I would expect is the implicit ID for `Foo` becomes `FooID`… all `fooId` columns become `fooID`.

Moving from ‘ID’ to ‘Id’ is losing valuable casing information & if it weren’t important, we wouldn’t be bothering with upper case letters at all in English (& other languages with bicameral scripts). …It is also a bit [Freudian](https://en.wikipedia.org/wiki/Id,_ego_and_superego#Id) to use “Id”.

**Comment by parsonsmatt** [comment]
Yeah, that assumption is baked into a lot of the TH code. I don't anticipate changing that to be easy or particularly flexible.

**Comment by parsonsmatt** [comment]
While you can define the `PersistEntity` stuff yourself, it's not really easy or well-supported.
